helix-core/src/main/java/org/apache/helix/controller/GenericHelixController.java

```diff
@@ -129,13 +129,19 @@
  * 4. select the messages that can be sent, needs messages and state model constraints <br>
  * 5. send messages
  */
-public class GenericHelixController implements IdealStateChangeListener, LiveInstanceChangeListener,
-                                               MessageListener, CurrentStateChangeListener,
+public class GenericHelixController implements IdealStateChangeListener,
+                                               LiveInstanceChangeListener,
+                                               MessageListener,
+                                               CurrentStateChangeListener,
                                                TaskCurrentStateChangeListener,
                                                CustomizedStateRootChangeListener,
                                                CustomizedStateChangeListener,
-    CustomizedStateConfigChangeListener, ControllerChangeListener,
-    InstanceConfigChangeListener, ResourceConfigChangeListener, ClusterConfigChangeListener {
+                                               CustomizedStateConfigChangeListener,
+                                               ControllerChangeListener,
+                                               InstanceConfigChangeListener,
+                                               ResourceConfigChangeListener,
+                                               ClusterConfigChangeListener,
+                                               ClusterPauseChangeListener {
```

Comment:
This list is becoming super long, can we have a ClusterManagementStatusChangeListener and serve for all the status paths, like Pause, ClusterPause, and Maintenance.

Reply:
Only ClusterPause needs a dataChange listener for exiting pause, because controller needs to wait for participants fully exited. The other 2: controller pause and maintenance don't need the data change listener, as they are simply deleted for exiting.